apis/nucleus/src/components/listbox/components/ListBoxSearch.jsx

```diff
@@ -7,46 +7,68 @@ import InstanceContext from '../../../contexts/InstanceContext';
 
 const TREE_PATH = '/qListObjectDef';
 
-export default function ListBoxSearch({ model, keyboard, dense = false, visible = true }) {
+export default function ListBoxSearch({ selections, model, keyboard, dense = false, visible = true }) {
   const { translator } = useContext(InstanceContext);
   const [value, setValue] = useState('');
-  const [isSearching, setIsSearching] = useState(false);
   const theme = useTheme();
 
+  const cancel = () => selections.isActive() && selections.cancel();
+
   const abortSearch = async () => {
-    await model.abortListObjectSearch(TREE_PATH);
-    setIsSearching(false);
+    if (!selections.isModal()) {
+      return;
+    }
+    try {
+      await model.abortListObjectSearch(TREE_PATH);
+    } finally {
+      setValue('');
+    }
   };
 
   useEffect(() => {
-    if (!visible && isSearching) {
-      abortSearch();
+    if (!visible) {
+      cancel(); // cancel and abort when toggling off search
     }
   }, [visible]);
 
-  const onChange = (e) => {
+  useEffect(() => {
+    selections.on('deactivated', abortSearch);
+    return () => {
+      selections.removeListener && selections.removeListener('deactivated', abortSearch);
+    };
+  }, []);
+
+  const onChange = async (e) => {
     setValue(e.target.value);
-    if (e.target.value.length === 0) {
-      abortSearch();
-    } else {
-      model.searchListObjectFor(TREE_PATH, e.target.value);
-      setIsSearching(true);
+    if (e.target.value.length) {
+      return model.searchListObjectFor(TREE_PATH, e.target.value);
     }
+    return undefined;
```

Comment:
why do onChange no longer do anything when searchstring is changed to empty? (it feels wrong)

Reply:
We don't want to do anything when the search string is empty.
The `abortSearch` call for empty strings was introduced as a quick fix this summer (not sure who did that). To achieve feature parity with the current Listbox, we should not call abort search in this case. We should only do that when deactivating selections actively or passively, e.g. through clicking outside the viz or pressing Esc.

Reply:
the sense-client do call abortListObjectSearch in this case.
how does the feature parity with sense-client work without it?

Reply:
Testing the the current selection listbox popover in nebula serve. I can not stop searching without closing the popover and reopening it. (just removing the search string keeps serching for the last letter removed)

Reply:
I would expect is to also be possible to remove the search string to end the search. (matching how it works in sense-client)

Reply:
@T-Wizard How do you search with an empty string in a Filterpane on Stage?

Reply:
For the records, the case referred to here is when:
1. performing a search
2. clearing the search (no submit needed)

I have adapted the code to accomodate for this case too. Thanks @T-Wizard for finding this.